cmd/iavlserver/main.go

```diff
@@ -0,0 +1,191 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"runtime/pprof"
+	"syscall"
+
+	"github.com/gogo/gateway"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	dbm "github.com/tendermint/tm-db"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/grpclog"
+
+	pb "github.com/cosmos/iavl/proto"
+	"github.com/cosmos/iavl/server"
+)
+
+var (
+	dbDataDir       = flag.String("datadir", "", "The database data directory")
+	dbName          = flag.String("db-name", "", "The database name")
+	dbBackend       = flag.String("db-backend", string(dbm.GoLevelDBBackend), "The database backend")
+	version         = flag.Int64("version", 0, "The IAVL version to load")
+	cacheSize       = flag.Int64("cache-size", 10000, "Tree cache size")
+	gRPCEndpoint    = flag.String("grpc-endpoint", "localhost:8090", "The gRPC server endpoint (host:port)")
+	gatewayEndpoint = flag.String("gateway-endpoint", "localhost:8091", "The gRPC-Gateway server endpoint (host:port)")
+	cpuProfile      = flag.String("cpuprofile", "", "If set, write CPU profile to this file")
+	memProfile      = flag.String("memprofile", "", "If set, write memory profile to this file")
```

Comment:
Maybe we could simply expose the pprof server via HTTP instead of using command-line arguments for profiling? This allows the caller to better control the profiling by submitting options.